org.adempiere.ui/src/org/compiere/apps/AbstractProcessCtl.java

```diff
@@ -290,6 +313,34 @@ public void run ()
 		}			//	*** Process submission ***
 	}   //  run
 
+	/**
+	 * Find Print Format for Jasper Report
+	 * @param pi
+	 * @param pinstance
+	 * @return AD_PrintFormat_ID or -1 if not found
+	 */
+	private int findPrintFormatForJasper(ProcessInfo pi, MPInstance pinstance) {
+		// from PInstance
+		if(pinstance.getAD_PrintFormat_ID() > 0) {
+			return pinstance.getAD_PrintFormat_ID();
+		}
+		// from Process Info
+		if(pi.getSerializableObject() != null && pi.getSerializableObject() instanceof MPrintFormat) {
+			return ((MPrintFormat) pi.getSerializableObject()).getAD_PrintFormat_ID();
+		}
+		else if(pi.getTransientObject() != null && pi.getTransientObject() instanceof MPrintFormat) {
+			return ((MPrintFormat) pi.getTransientObject()).getAD_PrintFormat_ID();
+		}
+		// find for table and process
+		if(pi.getAD_Process_ID() > 0 && pinstance.getAD_Table_ID() > 0) {
+			String sqlWhere = "JasperProcess_ID = ? AND AD_Table_ID = ?";
+			return new Query(Env.getCtx(), MPrintFormat.Table_Name, sqlWhere, null)
+					.setParameters(pi.getAD_Process_ID(), pinstance.getAD_Table_ID())
+					.firstId();
+		}
```

Comment:
If jasper report is running from a print format, that print format must have been saved into either AD_PInstance or ProcessInfo, don't think there's need to check via AD_Table_ID as well.

Reply:
I see thanks
I've removed the unnecessary code & marked the PR as ready for review